DOCS-2703: Add support for camera with esp32 (#3294)
viamrobotics/docs: docs/installation/micro-rdk-dev.md

Navigate to [the Viam app](https://app.viam.com) and [add a new machine](/cloud/machines/#add-a-new-machine) in your desired location.

Click on the name of the machine to go to the machine's page.

> Then, navigate to the **CONFIGURE** tab and [configure your machine](/configure/) with an `esp32` board and any components you want.
> You can also do this later.

Reviewer: I actually don't think they should do that at this stage. Once connected they can actually test. So makes more sense to complete setup first. 
```suggestion
```

```

Select `templates/project` when prompted.
Give the project a name of your choice.
Select `esp32` for **MCU**.
If you wish to configure an `esp32-camera` or a `fake` camera as a component of your machine, select **true** for **include camera module and traits**.

You will be prompted to paste your machine's `viam-server` robot JSON configuration into the terminal, which is the same thing as its machine cloud credentials.

To obtain this:

> - Navigate to [your new machine's](#create-a-new-machine) page on [the Viam app](https://app.viam.com) and select the **CONFIGURE** tab.

Reviewer: use this one, it's got the updated button
```
{{<imgproc src="configure/machine-part-info.png" resize="500x" declaredimensions=true alt="Restart button on the machine part info dropdown">}}
```

Author: changed

{{% snippet "secret-share.md" %}}

> ### Add your machine cloud credentials as a file in your project

Author: @mattjperez found this necessary but seems like a bug-- how does micro-RDK team want to address?

Reviewer 2: made a [ticket](https://viam.atlassian.net/browse/RSDK-8623?atlOrigin=eyJpIjoiY2VlOGQ2N2M0MjVhNDkwYzhkMWJlNzgwMDM2NDM4MWEiLCJwIjoiaiJ9) and PR. I reworded the prompt to say machine cloud config like what's in UI, but will need to see what others think.

Reviewer 2: fix merged this morning. this section should no longer be needed, however I'd include checking the viam.json file in general troubleshooting steps

Author: awesome, sounds great thanks!

Navigate to the directory of the project you just created.
Create a new <file>viam.json</file> file and paste the same `viam-server` machine cloud credentials in.
Save the file.

> You can do this on the command line like this:
> 
> ```sh { class="command-line" data-prompt="$"}
> cd <your-path-to/your-project-directory>
> touch viam.json
> vim viam.json
> ```
> 
> Replace `vim` with whatever text editor you prefer.

Reviewer: we'll leave that to the reader. If they don't know how to do this, they won't be able to develop with rust I think
```suggestion
```

### Upload `viam-micro-server` to your ESP32

> Now, upload the project to connect to your ESP32 and remotely control it live on [the Viam app](https://app.viam.com):

Reviewer: ```suggestion
Now, flash the project to your ESP32 and it will connet to [the Viam app](https://app.viam.com) from which you can then remotely control it:
```

{{% /tab %}}
{{% /tabs %}}

When prompted, select the serial port that your ESP32 is connected to through a data cable.

If successful, you will retain a serial connection to the board until you press `Ctrl-C`.
To manage this connection, consider running it within a dedicated terminal session, or under `tmux` or `screen`.

> While the serial connection is live, you can also restart the currently flashed image with `Ctrl-R`.

Reviewer: On line 307: where do they see the last online, let's add more specificity or less. "If successful, you machine's page will show that your machine is **Live**."

Author: added

> ### Troubleshooting

Reviewer: nice

If you run into the error `Failed to open serial port` when flashing your ESP32 with Linux, make sure the user is added to the group `dialout` with `sudo gpasswd -a $USER dialout`.

> If you get the following error while connecting to your esp32:

Author: ```suggestion
If you get the following error while connecting to your ESP32:
```
